sleap_nn/training/model_trainer.py

```diff
@@ -68,38 +86,112 @@ def __init__(self, config: OmegaConf):
         self.steps_per_epoch = self.config.trainer_config.steps_per_epoch
 
         # initialize attributes
-        self.model_type = None
         self.model = None
         self.provider = None
         self.skeletons = None
         self.train_data_loader = None
         self.val_data_loader = None
 
+        # check which head type to choose the model
+        for k, v in self.config.model_config.head_configs.items():
+            if v is not None:
+                self.model_type = k
+                break
+
+        if not self.config.trainer_config.save_ckpt_path:
+            self.dir_path = "."
+        else:
+            self.dir_path = self.config.trainer_config.save_ckpt_path
+
+        if not Path(self.dir_path).exists():
+            try:
+                Path(self.dir_path).mkdir(parents=True, exist_ok=True)
+            except OSError as e:
+                print(
+                    f"Cannot create a new folder. Check the permissions to the given Checkpoint directory. \n {e}"
+                )
+
         # set seed
         torch.manual_seed(self.seed)
 
+    def _get_data_chunks(self, func, train_labels, val_labels):
+        """Create a new folder with pre-processed data stored as `.bin` files."""
+        ld.optimize(
+            fn=func,
+            inputs=[(x, train_labels.videos.index(x.video)) for x in train_labels],
+            output_dir=(Path(self.dir_path) / "train_chunks").as_posix(),
+            num_workers=self.config.trainer_config.train_data_loader.num_workers,
+            chunk_size=(
+                self.config.data_config.chunk_size
+                if "chunk_size" in self.config.data_config
+                and self.config.data_config.chunk_size is not None
+                else 100
```

Comment:
Leave a TODO comment here to refactor this once #75 is done. Defaults should be handled in the config validation.